### ISSUE TITLE
Poll → Scroll To Bottom

### ISSUE THREAD

**Issue by KevinDuganJr** [YAFNET/YAFNET#959]
Using the latest NETFX commit code. When voting in a poll you are thrown to the last post. 

**Feature Request.**
Can the scrolling to bottom be handled via link handling rather than every action/event forcing a scroll to bottom?

I believe YAF used to handle scroll to bottom or maybe it was called "go to post"  via URL → ~/forum/posts/m459103-packers#post459103

**Comment by w8tcha** [comment]
This is was only happen in the sample application. I committed a fix to it.

**Comment by w8tcha** [comment]
> Can the scrolling to bottom be handled via link handling rather than every action/event forcing a scroll to bottom?

Why, there is zero reason in my opinion. If click on a topic I would prefer going to the newest post.

If I understand you, you would prefer always load the first page of the topic and don't go to a specific post?

For me that makes no sense. What would be the intention?

**Comment by KevinDuganJr** [comment]
> > Can the scrolling to bottom be handled via link handling rather than every action/event forcing a scroll to bottom?
> 
> Why, there is zero reason in my opinion. If click on a topic I would prefer going to the newest post.
> 
> If I understand you, you would prefer always load the first page of the topic and don't go to a specific post?
> 
> For me that makes no sense. What would be the intention?

When a user clicks on a Topic, the expectation is for the topic to be opened to the first page. It's no different than opening up a book. If you opened up a book and all the pages flipped by to the last chapter, that would not make sense either.

Example. I click on the topic and my screen provides a jarring experience by scrolling to the bottom. Now I have to scroll all the way back up to read the first post to see the organic flow of the discussion.
![image](https://github.com/YAFNET/YAFNET/assets/5612416/b3e0ef62-e41d-44c9-a099-1e10d5e372e3)

I'll use YAF Support Forum for another example. I click on the title of the topic. My expectation is to be brought to the first page, first post. Instead my screen scrolls on it's own all the way down to the bottom of the last page to the last post. 
![image](https://github.com/YAFNET/YAFNET/assets/5612416/f37e795f-5102-43f4-9b9b-b324b4288818)



When the user is expecting (or at least reasonably expecting) the last post to be shown is when they click on a link as shown below.
![image](https://github.com/YAFNET/YAFNET/assets/5612416/daa17760-7147-4460-bab3-41eb701d7fd2)


I'm not sure why you would want the newest post when you click on a topic. It's a discussion forum, so don't you want to see how the discussion got to that last post? 

As for auto scrolling in general. It's against most user experience best practices. Me personally? I have always despised my screen scrolling with out my intervention. Especially when I open up a webpage and start to navigate and a second later my page scrolls. There's latency that has to be considered when you're auto "anything" to a user design to ensure a consistent and pleasurable user experience. 

YAF previously offered up multiple options and none of which auto scrolled, but just BOOM, there you go, haha.
If you clicked the Topic, first page, first post. 
If you clicked "Latest Post in Topic", you'd be brought to last page, last topic.
If you clicked "Newest Post in Topic", you'd be brought to the post that was newest since your last visit.

I'm paraphrasing on the naming here, but you get the idea. The user had control, not the website and that's how I feel it should be as that is what makes sense to me as a user. 

All in all I do want to point something out. The software feels a bit smoother with this last update and THANK YOU SO MUCH for the BBCode Editor! haha.

**Comment by w8tcha** [comment]
Okay i found the problem. It should always go to the first unread post, but instead it was going to the last.

> When a user clicks on a Topic, the expectation is for the topic to be opened to the first page. It's no different than opening up a book. If you opened up a book and all the pages flipped by to the last chapter, that would not make sense either.

I get that, but it would also not make  sense to always open a book on the first page like you want, I would want to go to the page where i stop reading. Like what yaf is now correctly doing.

**Comment by w8tcha** [comment]
i added a new setting, the scrolling can now be disabled.

**Comment by KevinDuganJr** [comment]
> Okay i found the problem. It should always go to the first unread post, but instead it was going to the last.
> 
> > When a user clicks on a Topic, the expectation is for the topic to be opened to the first page. It's no different than opening up a book. If you opened up a book and all the pages flipped by to the last chapter, that would not make sense either.
> 
> I get that, but it would also not make sense to always open a book on the first page like you want, I would want to go to the page where i stop reading. Like what yaf is now correctly doing.

Yes, in your scenario, yes, a bookmark of sorts. I was referring to the first time you open up a book, not the times after that. I should have clarified that -- my mistake. But it sounds like you got the overall idea anyway.

**Comment by KevinDuganJr** [comment]
> i added a new setting, the scrolling can now be disabled, i hope this resolves the issue

I will have to give that a try. I'm trying to figure out what I did that is causing me to need to login everyday now. I think my cookie timeout might have been reset when I cleared out my registry settings. lol oops!! 😁

**Comment by w8tcha** [comment]
I also removed the actually scrolling to the post, now it goes directly to the post.

**Comment by KevinDuganJr** [comment]
> I also removed the actually scrolling to the post, now it goes directly to the post.

Nice. I had some issues when I tried upgrading this morning so I had to roll back to last week. I'm starting over to see what got hosed. I'll report any issue, but I think it's with my Registry because I cannot save the Admin/Settings without it blowing up with the NULL error below. So, gonna run YAF standalone and see if I can reproduce the issue. 

![image](https://github.com/YAFNET/YAFNET/assets/5612416/5f5b4522-0ac4-48df-a608-d423ef8a3b2d)
